### ISSUE TITLE
Bitcoin "Bounty/Payment/Reward" System for Copay Improvements?

### ISSUE THREAD

**Issue by KeyJockey** [bitpay/copay#3204]
Hey guys, any possibility of implementing a reward/bounty for certain improvements or bug fixes in Copay wallet here?

It's just an idea but srsly I'd be willing to actually PAY something to get CERTAIN THINGS in Copay actually DONE, rather than having to wait until enough people agree to do them and/or until the volunteer open source community finds time to work on 'em.

I'm using Copay every day to actually get work and business done, it's an important TOOL for me and I really like it so far, despite the shortcomings it's still the best wallet out there *but still* it needs some stuff *fixed* pronto!

And before someone says "fork it" to me, I already answered that elsewhere: It doesn't serve everyone's best interest to have a million versions of this wallet, I wanna see ONE BEST WALLET software standardized that I'll be able to recommend to *everyone*... and also not have to endlessly maintain a private version of it.

Bottom line: I'd simply be willing to sometimes PAY something to get what I want, in THIS wallet software, rather than have to only always wait and hope that it might, maybe, possibly, who knows, get done "some day"...

**Comment by dan-da** [comment]
I like this idea.  

Also, if the core dev team itself doesn't have bandwidth to setup/manage a bounty system, I don't see any reason why a third party could not do it.

**Comment by szhongben** [comment]
Synapsi.com enables you to contribute and collect bitcoin bounties for GitHub issues using multi-sig escrow.

**Comment by dabura667** [comment]
Go to irc freenode #nodejs and state you have a bounty, post a link here and see what you get.

I don't see why a "system" needs to be in place to facilitate trade between two people.

If you want a bounty, make an offer and post it all over the place.

**Comment by KeyJockey** [comment]
Well, as you may have noticed from my post here and on the BTCTalk Copay page, I *really* want Copay to have a manual over-ride for the miner fee rather than being *forced* to accept the dynamically calculated fee the way it now works, after version 1.1.2.  -- See my issue https://github.com/bitpay/copay/issues/3111

Not having control over precise miner fee basically makes Copay un-useable for me even though I really like a lot of the OTHER new features.  This is seriously annoying LOL

So: what would this "COST" actually?  

HOW much bounty (bribe LOL) would be needed to actually get this done, esp considering the devs here have tagged my 3111 issue "low prio"?  :(  

Depending on how cheap or expensive it is (relatively speaking) I may be willing to cover the whole cost myself -or- at least try to get a pool going, to get it done, IF anyone else around here actually might agree with me that this is seriously annoying...

**Comment by dabura667** [comment]
> what would this "COST" actually

If you hired some node developer off the street it might take them a day or two to do the actual fix, and maybe longer to build you a working app.

I would ballpark a for-hire guy doing it maybe for 2-3 grand. So 10 BTC sounds about right.

However, just because you paid some guy to implement it does not require the Copay team to include it.

You could build your own custom Copay app and install it directly from apk file for Android, and maybe testflight if you have iPhone.

**Comment by KeyJockey** [comment]
LOL well it ain't worth THAT much to me.  And yeah as I said elsewhere I don't want to FORK the code for this, I wanna see it done in actual Copay.  I want COPAY to be the BEST wallet that I can recommend to *everyone* to use, and IMHO for it to be back on track to being that, it absolutely NEEDS to have a manual override option of the miner fee value for advanced users!

Srsly what's so bad about this idea? What's the downside to letting advanced users manually set their preferred transaction fee?  Yeah sure OK, I see the coolness of the dynamic fee feature, okay fine, but why does it have to be forced down everyone's throat for EVERY transaction?? I don't get it why so few people seem to be backing me up on this here?

So how about it, Copay Devs?
Hey I'm offering ya'll a bribe here!  
How much $$$ do you guys want to put this in the next version???  LOL

**Comment by KeyJockey** [comment]
Hey guys FYI looks like szhongben's right & he's added a bounty to my :3111 issue here, using Synapsi.

Pretty cool and looks like it might fit the bill here fine BUT my only question is HOW or "who" decides that the "question" (i.e. copay bug-fix or feature added) is finally correctly "answered" to release the bounty funds from Synapsi escrow?

I just wanna be sure I understand this fully before committing bitcoin funds to the escrow: any chance someone could run a scam here by exploiting this system -OR- is it pretty foolproof or secure already?

IF this already works fine, prob can close this feature request... LOL and sorry I wasn't aware of Synapsi before posting it in the first place :smiley:

**Comment by szhongben** [comment]
You need to decide if the issue has been resolved before releasing funds from escrow. Your funds are secure because both you and Synapsi need to sign any transaction. It's not foolproof in that Copay needs to trust that you won't request a refund (reputation can matter) if they resolve the issue.

**Comment by matiu** [comment]
Thanks for the suggestion. There are many bounty systems for github already any of those can be used with Copay.
